nf_core/pipelines/lint/rocrate_readme_sync.py

```diff
@@ -0,0 +1,68 @@
+import json
+import logging
+from pathlib import Path
+
+log = logging.getLogger(__name__)
+
+
+def rocrate_readme_sync(self):
+    """
+    Check if the RO-Crate description in ro-crate-metadata.json matches the README.md content.
+    If the --fix is set, the RO-Crate description will be updated to match the README.md content.
+    """
+
+    passed = []
+    failed = []
+    ignored = []
+    fixed = []
+    could_fix: bool = False
+
+    # Check if the file exists before trying to load it
+    metadata_file = Path(self.wf_path, "ro-crate-metadata.json")
+    readme_file = Path(self.wf_path, "README.md")
+
+    # Only proceed if both files exist
+    if not (metadata_file.exists() and readme_file.exists()):
+        if not metadata_file.exists():
+            ignored.append("`ro-crate-metadata.json` not found")
+        if not readme_file.exists():
+            ignored.append("`README.md` not found")
+        return {"passed": passed, "failed": failed, "ignored": ignored}
+
+    try:
+        metadata_content = metadata_file.read_text(encoding="utf-8")
+        metadata_dict = json.loads(metadata_content)
+    except json.JSONDecodeError as e:
+        log.error("Failed to decode JSON from `ro-crate-metadata.json`: %s", e)
+        ignored.append("Invalid JSON in ro-crate-metadata.json")
+        return {"passed": passed, "failed": failed, "ignored": ignored}
+    readme_content = readme_file.read_text(encoding="utf-8")
+    graph = metadata_dict.get("@graph")
+    if not graph or not isinstance(graph, list) or not graph[0] or not isinstance(graph[0], dict):
+        ignored.append("Invalid RO-Crate metadata structure.")
+    else:
+        # Check if the 'description' key is present
+        if "description" not in graph[0]:
+            ignored.append(
+                "No description found in ro-crate-metadata.json, add a description to the RO-Crate metadata."
+            )
+            graph[0]["description"] = readme_content
```

Comment:
This way we only add a description when `--fix` is specified
```suggestion
            if "rocrate_readme_sync" in self.fix:
                graph[0]["description"] = readme_content
            else:
                return {"passed": passed, "failed": failed, "ignored": ignored}
```